src/+datools/+statistical/+ensemble/EnF.m

```diff
@@ -1,34 +1,35 @@
-classdef EnF < handle
-    % This is the base class for all statistical methods
-    % Derive from this class and implement methods/functions as required
-    % Deriving from handle base class allows an object of this class to be
-    % passed by reference.
+classdef EnF < datools.DABase
+%ENF This is the base class for all statistical methods
+%   Derive from this class and implement methods/functions as required
+%   Deriving from handle base class allows an object of this class to be
+%   passed by reference.
 
     properties
-        Model % type of ODE solver (ode45/Runge Kutta) and the model (eg: Lorenz63)
-        ModelError % type err
-        Observation % type of obervation
-        Ensemble % current ensemble values for all the states
-        Weights % Weight of each particle/ensemble
-        Inflation % inflation constant
-        Rejuvenation % rejuvenation constant
-        Localization % A boolean if localization needs to be used
-        Parallel % A boolean if parallel threads are to be implemented
-        RankHistogram % State variables for which RH is needed
-        RankValue % store the RH values for each state variables required
-        ResamplingThreshold % threshold below which resampling needs to be done
+        %Model                 % type of ODE solver and the model 
+        %ModelError            % type err
+        %Observation           % type of obervation
+        Ensemble               % current ensemble values for all the states
+        Anomalies              % Anomalies
+        Weights                % Weight of each particle/ensemble
+        Inflation              % inflation constant
+        Rejuvenation           % rejuvenation constant
+        Localization           % A boolean if localization needs to be used
+        Parallel               % A boolean if parallel threads are used
+        RankHistogram          % State variables for which RH is needed
+        RankValue              % store the RH values for each state variables required
```

Comment:
I still don't think RankValue should be in the filter class. We should perhaps have it be an external thing, or passed as a hook or something.

Reply:
I agree. Sorry I did not mention that I was planning to remove rank histogram from statistical class and keep it independent.